### PR TITLE
refactor(hex): Datasource schema validation

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

- Add schema validation for hex datasource

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/hex/schema.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { z } from 'zod';
+import { LooseArray } from '../../../util/schema-utils';
+import type { Release, ReleaseResult } from '../types';
+
+export const HexRelease = z
+  .object({
+    html_url: z.string().optional(),
+    meta: z
```
Comment (rarkins) [inline lib/modules/datasource/hex/schema.ts:8]: `meta` is a confusing name to retain for this URL string

File: lib/modules/datasource/hex/schema.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+          if (releaseTimestamp) {
+            release.releaseTimestamp = releaseTimestamp;
+          }
+
+          return release;
+        })
+    ).refine((releases) => releases.length > 0, 'No releases found'),
+  })
+  .transform(
```
Comment (rarkins) [inline lib/modules/datasource/hex/schema.ts:34]: I find it a little strange how you use transform in both the first part of this parsing as well as here - there doesn't seem to be a clear separation between validating/parsing the data and then transforming it

File: lib/modules/datasource/hex/schema.ts (hunk 1 above)
Comment (viceice) [reply lib/modules/datasource/hex/schema.ts:8]: this is coming from the json server result, so we can't change it here

Comment (rarkins) [reply lib/modules/datasource/hex/schema.ts:8]: We are flattening it though. It makes sense as an object but not as a string

Comment (rarkins) [reply lib/modules/datasource/hex/schema.ts:8]: @zharinov can this be left as meta.links.GitHub (optional) during the validation and only transformed later?

**Review by viceice — COMMENTED** [review]
you're not following your own guide and transform while define

File: lib/modules/datasource/hex/schema.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        }),
+      })
+      .nullable()
+      .catch(null),
+    releases: LooseArray(
+      z.object({
+        version: z.string(),
+        inserted_at: z.string().optional(),
+      }),
+    ).refine((releases) => releases.length > 0, 'No releases found'),
```
Comment (viceice) [inline lib/modules/datasource/hex/schema.ts:21]: what is this doing? 🤔

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/modules/datasource/hex/schema.ts (hunk 3 above)
Comment (zharinov) [reply lib/modules/datasource/hex/schema.ts:21]: We need to return `null` for empty array of releases (probably, never happens)

**Review by viceice — APPROVED** [review]

**Review by secustor — APPROVED** [review]

**Pull request merged**